### DISCUSSION TITLE
Vvenc scaling beyond threads

### DISCUSSION THREAD

**Discussion by pbudhkar** [fraunhoferhhi/vvenc#139] [Q&A]
Hi Maintainers,
I want to understand the reason why Vvenc runtime is not scalable beyond 8 threads. I am running it on intel icelake machine and the machine is greatly underutilized. Is the runtime bounded by the algorithm or implementation? 

More details as follows:
The machine has 128 logical cores and of which only ~7% cores are utilized. My command looks like 
./bin/relwithdebinfo-static/vvencFFapp -i KristenAndSara_1280x720_60.yuv -s 1280x720 -fr 60 -c ./cfg/randomaccess_fast.cfg --QP 22 -t 32 -v 3 -b str.266.
Any help is appreciated.

**Comment by adamjw24** [comment]
Well, in your video, thre just isn't some much workload to parallelize. 720p is just 6 CTU lines (for CTU128, which is used in medium), even with frame parallelization its not really many independent workloads to be processed. You could try and reduce the CTU size (`--CTUSize=64`).

Have a look at other options for multi-threading: https://github.com/fraunhoferhhi/vvenc/wiki/Encoder-Performance#multi-threading

`--WaveFronySynchro=1` might help. Also tiling might help (e.g. `--Tiles=2x1`), especially combined with CTU size reduction and wavefront processing.

There is a reason we chose 8 threads as the default. We are not looking to make VVenC realtime capable, so we want to have optimal utilization of some threads, not necessarily all. On a 64 core machine you might rather split the video into chunks and encode those in parallel.
